systemtest/src/test/java/io/strimzi/systemtest/security/oauth/OauthPlainST.java

```diff
@@ -69,6 +71,95 @@ void testProducerConsumer() {
+                .withBootstrapAddress(KafkaResources.plainBootstrapAddress(oauthClusterName))
+                .withTopicName(TOPIC_NAME)
+                .withMessageCount(MESSAGE_COUNT)
+                .withOAuthClientId(OAUTH_CLIENT_AUDIENCE_PRODUCER)
+                .withOAuthClientSecret(OAUTH_CLIENT_AUDIENCE_SECRET)
+                .withOAuthTokenEndpointUri(keycloakInstance.getOauthTokenEndpointUri())
+                .build();
+
+        KafkaOauthExampleClients oauthInternalClientConsumerJob = new KafkaOauthExampleClients.Builder()
+                .withConsumerName(OAUTH_CLIENT_AUDIENCE_CONSUMER)
+                .withBootstrapAddress(KafkaResources.plainBootstrapAddress(oauthClusterName))
+                .withTopicName(TOPIC_NAME)
+                .withMessageCount(MESSAGE_COUNT)
+                .withOAuthClientId(OAUTH_CLIENT_AUDIENCE_CONSUMER)
+                .withOAuthClientSecret(OAUTH_CLIENT_AUDIENCE_SECRET)
+                .withOAuthTokenEndpointUri(keycloakInstance.getOauthTokenEndpointUri())
+                .build();
+
+        oauthInternalClientProducerJob.createAndWaitForReadiness(oauthInternalClientProducerJob.producerStrimziOauthPlain().build());
+        ClientUtils.waitForClientSuccess(OAUTH_CLIENT_AUDIENCE_PRODUCER, NAMESPACE, MESSAGE_COUNT);
+
+        oauthInternalClientConsumerJob.createAndWaitForReadiness(oauthInternalClientConsumerJob.consumerStrimziOauthPlain().build());
+        ClientUtils.waitForClientSuccess(OAUTH_CLIENT_AUDIENCE_CONSUMER, NAMESPACE, MESSAGE_COUNT);
+
+        JobUtils.deleteJobWithWait(NAMESPACE, OAUTH_CLIENT_AUDIENCE_PRODUCER);
+        JobUtils.deleteJobWithWait(NAMESPACE, OAUTH_CLIENT_AUDIENCE_CONSUMER);
+
+        LOGGER.info("Revert Kafka configuration to original one.");
+        KafkaResource.replaceKafkaResource(oauthClusterName, kafka -> {
+            kafka.getSpec().getKafka().setListeners(new ArrayOrObjectKafkaListenersBuilder()
+                .addNewGenericKafkaListener()
+                    .withName(Constants.PLAIN_LISTENER_DEFAULT_NAME)
+                    .withPort(9092)
+                    .withType(KafkaListenerType.INTERNAL)
+                    .withTls(false)
+                    .withNewKafkaListenerAuthenticationOAuth()
+                        .withValidIssuerUri(keycloakInstance.getValidIssuerUri())
+                        .withJwksExpirySeconds(keycloakInstance.getJwksExpireSeconds())
+                        .withJwksRefreshSeconds(keycloakInstance.getJwksRefreshSeconds())
+                        .withJwksEndpointUri(keycloakInstance.getJwksEndpointUri())
+                        .withUserNameClaim(keycloakInstance.getUserNameClaim())
+                        .withEnablePlain(true)
+                        .withTokenEndpointUri(keycloakInstance.getOauthTokenEndpointUri())
+                    .endKafkaListenerAuthenticationOAuth()
+                .endGenericKafkaListener()
+                .build());
+        });
```

Comment:
I think you should wait here for RU and Kafka readiness

Reply:
You mean only `KafkaUtils.waitForKafkaReady(oauthClusterName);`? Or anything else?

Reply:
if RU is needed for those changes, you should use something like `StatefulsetUtils.waitForRollingUpdate` and then check status with `KafkaUtils.waitForKafkaReady(oauthClusterName);` to avoid races